### ISSUE TITLE
Using persist_id for committing the configuration

### ISSUE THREAD

**Issue by muffizone** [ncclient/ncclient#357]
The use of `persist_id` is currently not supported for committing the configuration from a different session from the one where the confirmed commit was issued. 

Snippet from the netconf RFC

```
   If the <persist> element is given in the confirmed <commit> operation, a
   follow-up commit and the confirming commit can be given on any
   session, and they MUST include a <persist-id> element with a value
   equal to the given value of the <persist> element.
```

XML request from the RFC
```
     <!-- confirm the persistent confirmed-commit,
          possibly from another session -->
     <rpc message-id="102"
          xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
       <commit>
         <persist-id>IQ,d4668</persist-id>
       </commit>
     </rpc>
```

To implement this feature, I'd suggest a small change to the `request` method in the `Commit` class.

```
class Commit(RPC):
    "`commit` RPC. Depends on the `:candidate` capability, and the `:confirmed-commit`."

    DEPENDS = [':candidate']

    def request(self, confirmed=False, timeout=None, persist=None, persist_id=None):
        """Commit the candidate configuration as the device's new current configuration. Depends on the `:candidate` capability.

        A confirmed commit (i.e. if *confirmed* is `True`) is reverted if there is no followup commit within the *timeout* interval. If no timeout is specified the confirm timeout defaults to 600 seconds (10 minutes). A confirming commit may have the *confirmed* parameter but this is not required. Depends on the `:confirmed-commit` capability.

        *confirmed* whether this is a confirmed commit

        *timeout* specifies the confirm timeout in seconds

        *persist* make the confirmed commit survive a session termination, and set a token on the ongoing confirmed commit

        *persist_id* value must be equal to the value given in the <persist> parameter to the previous <commit> operation.
        """
        node = new_ele("commit")
        if confirmed:
            self._assert(":confirmed-commit")
            sub_ele(node, "confirmed")
            if timeout is not None:
                sub_ele(node, "confirm-timeout").text = timeout
            if persist is not None:
                sub_ele(node, "persist").text = persist
+        if persist_id is not None:
+            sub_ele(node, "persist-id").text = persist_id

        return self._request(node)
```

Let me know if everyone is fine with this change and I can create the merge request with the above changes.

**Comment by einarnn** [comment]
@muffizone, I think you have misinterpreted  the RFC. The persistence id is provided as the value of the `<persist>` element in the commit, and this value may already be passed in to the commit. If there is a later `<cancel-commit>`, then `<persist-id>` is used. this is how the current code behaves after a PR not that long ago.

Please reread the RFC and see if you agree.

**Comment by muffizone** [comment]
@einarnn 

How do we confirm the commit from a session which is different from the initial session where we did the confirmed commit operation? The ios-xr box i was testing on will not allow me to confirm the commit using the persist keyword but works with persist-id

**Comment by einarnn** [comment]
According to the RFC, the “persist” tag should be used when doing the initial commit and then the “persist-id”bath should be used when cancelling or confirming.

That doesn’t mean that the Cisco IOS XR team haven’t also misinterpreted the RFC. I will reach out to them and also to the RFC authors.

Cheers,

Einar

**Comment by muffizone** [comment]
I am talking about this particular snippet from the RFC which clearly states that `persist-id` element should be used while confirming or cancelling the commit from a different session.

```
If the <persist> element is given in the confirmed <commit> operation, a
follow-up commit and the confirming commit can be given on any
session, and they MUST include a <persist-id> element with a value
equal to the given value of the <persist> element.
```

The current cancel commit operation in ncclient also uses `persist-id` and not `persist` element for this.

**Comment by surajkhadekar** [comment]
This is my interpretation of RFC.  Any commit confirmed operation A can be confirmed from same session[session A]. if you want to commit this operation A from different session persist-id A is used to perform commit from different session [session B]

**Comment by einarnn** [comment]
Yeah, we need to make the persist-id available in the `<commit>` operation. Suggest the the code proposed initially should check for a client inadvertently specifying confirmed, persist and persist-id.

**Comment by einarnn** [comment]
@muffizone -- if you wish to modify your suggested patch as I suggest above, please feel free to submit a PR.

Cheers,

Einar

**Comment by einarnn** [comment]
Addressed in #358.
